Dockerfile

```diff
@@ -1,15 +1,52 @@
-FROM ubuntu:focal
+FROM python:3.9-slim
+
+LABEL org.opencontainers.image.source="hhttps://github.com/UST-QuAntiL/qhana-plugin-runner"
+
+# install git and remove caches again in same layer
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && apt-get install -y --no-install-recommends git && apt-get clean && rm -rf /var/lib/apt/lists/*
+
 WORKDIR /app
 
-RUN apt-get -y update && apt-get install -y python3 python3-distutils python3-apt wget
-RUN wget https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py
-RUN python3 get-poetry.py
-RUN rm get-poetry.py
-ENV PATH=${PATH}:/root/.poetry/bin
+RUN useradd gunicorn
+
+
+ENV FLASK_APP=qhana_plugin_runner
+ENV FLASK_ENV=production
+ENV PLUGIN_FOLDERS=/app/plugins:/app/extra-plugins:/app/git-plugins
+
+
+# can be server or worker
+ENV CONTAINER_MODE=server
+ENV DEFAULT_LOG_LEVEL=INFO
+ENV CONCURRENCY=2
+ENV CELERY_WORKER_POOL=threads
+
+# make directories and set user rights
+RUN mkdir /venv && mkdir --parents /app/instance &&  mkdir --parents /app/extra-plugins && mkdir --parents /app/git-plugins \
+    && chown --recursive gunicorn /app && chmod --recursive u+rw /app && chown --recursive gunicorn /venv
+
+# Wait for database
+ADD https://github.com/ufoscout/docker-compose-wait/releases/download/2.7.3/wait /wait
+RUN chmod +x /wait
+
+USER gunicorn
+
+# create and activate virtualenv as gunicorn user to allow installing additional dependencies later
+ARG VIRTUAL_ENV=/venv
+RUN python -m venv ${VIRTUAL_ENV}
+
+# change path to include virtualenv first (affects all following commands)
+ENV PATH="${VIRTUAL_ENV}/bin:$PATH"
+
+RUN python -m pip install --upgrade pip && python -m pip install gunicorn poetry invoke
+
+COPY --chown=gunicorn . /app
```

Comment:
Plugins are still copied from the plugins folder into the Docker Container. Is this intentional?

Reply:
No, plugins should be ignored as per dockerignore file.